File: src/stub/jatran/stub/TryCatchPatterns.scala
package jatran.stub

class TryCatchPatterns {
  def foo():Unit = {
    try {
    } finally {
    }
  }

  def bar():Unit = {
    try {
    } catch { 
      case e:Exception => 
    }
  }
  
  def baz():Unit = {
    try {
      bar()
    } catch {
      case e:Exception =>
    } finally {
      foo()
    }
  }
	
  def buzz():Unit = {
    try {
    } finally {
      baz()
    }
  }
  
}

File: src/stub/jatran/stub/TryCatchPatterns.java
package jatran.stub;

public class TryCatchPatterns {
	void foo() {
		try {
			
		} finally {
			
		}
	}
	
	void bar() {
		try {			
		} catch (Exception e)  {			
		}
	}
	
	void baz() {
		try {
			bar();
		} catch (Exception e) {
		} finally {
			foo();
		}
	}
	
	void buzz() {
		try {
			
		} finally {
			baz();
		}
	}
}
